packages/ag-charts-community/src/scene/node.ts

```diff
@@ -193,6 +193,21 @@ export abstract class Node {
         }
     }
 
+    protected sortChildren(compareFn?: (a: Node, b: Node) => number) {
+        this.dirtyZIndex = false;
+        if (!this.childNodes) return;
+
+        // Virtual children need to be sorted on the fly.
+        if (this.hasVirtualChildren()) return;
+
+        // Sort children, and re-add in new order (Set preserves insertion order).
+        const sortedChildren = [...this.childNodes].sort(compareFn);
```

Comment:
Would it be easier if we just kept `children` as an array? In theory, you couldn't have duplicate children, because you remove a child from its parent before adding it

Reply:
@iMoses any thoughts? This was originally an array, it's not clear to me if it's practically more useful to be an Array or a Set?

Reply:
If there's a good reason, and caching is a good reason, then I'm not against.
My main concern is that the implementation doesn't leak and for `children()` to remain a generator, preventing potentially copying very large arrays.